components/camel-jackson/src/main/java/org/apache/camel/component/jackson/AbstractJacksonDataFormat.java

```diff
@@ -157,7 +158,9 @@ public void marshal(Exchange exchange, Object graph, OutputStream stream) throws
         if (this.schemaResolver != null) {
             schema = this.schemaResolver.resolve(exchange);
         }
-        this.objectMapper.writerWithView(jsonView).with(schema).writeValue(stream, graph);
+        // using OutputStreamWriter because of Jackson not handling 4-byte characters properly with only a stream
+        // https://github.com/FasterXML/jackson-core/issues/223
+        this.objectMapper.writerWithView(jsonView).with(schema).writeValue(new OutputStreamWriter(stream), graph);
```

Comment:
Shouldn't you close the `OutputStreamWriter` here?

Reply:
Thanks Otavio. Please check the updated changes.